rest/route/host_spawn.go

```diff
@@ -1010,6 +1011,83 @@ func (h *hostExtendExpirationHandler) Run(ctx context.Context) gimlet.Responder
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/hosts/{host_id}/run_command
+//
+type hostRunScript struct {
+	sc  data.Connector
+	env evergreen.Environment
+
+	hostID string
+	script string
+}
+
+func makeHostRunScript(sc data.Connector, env evergreen.Environment) gimlet.RouteHandler {
+	return &hostRunScript{
+		sc:  sc,
+		env: env,
+	}
+}
+
+func (h *hostRunScript) Factory() gimlet.RouteHandler {
+	return &hostRunScript{
+		sc:  h.sc,
+		env: h.env,
+	}
+}
+
+func (h *hostRunScript) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+	h.hostID, err = validateID(gimlet.GetVars(r)["host_id"])
+	if err != nil {
+		return errors.Wrap(err, "invalid host id")
+	}
+
+	hostScript := model.APIHostScript{}
+	if err = util.ReadJSONInto(util.NewRequestReader(r), &hostScript); err != nil {
+		return errors.Wrap(err, "can't read host command from json")
+	}
+	h.script = hostScript.Script
+
+	return nil
+}
+
+func (h *hostRunScript) Run(ctx context.Context) gimlet.Responder {
+	user := MustHaveUser(ctx)
+	host, err := h.sc.FindHostByIdWithOwner(h.hostID, user)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "can't get host"))
```

Comment:
what about not found errors?

Reply:
the host not found case is handled by the connector.
https://github.com/evergreen-ci/evergreen/blob/a30d4c02a16df914ce3a5bf606a912b72df02d0d/rest/data/host.go#L49